source/views/sis/course-search/list.js

```diff
@@ -69,19 +71,23 @@ export class CourseSearchResultsList extends React.PureComponent<Props> {
 			title: 'Add Filters',
 			pathToFilters: ['courses', 'filters'],
 			onChange: filters => this.props.onFiltersChange(filters),
+			onLeave: filters => this.props.updateRecentFilters(filters),
```

Comment:
Any particular reason you added `onLeave`, instead of just updating the recent filters at the same time we call `onFiltersChange`?

Reply:
Yeah I figured we only want to add a filter combination to the recent filter combinations list after its been completed. For example, if a complete filter combination were `18/19:Fall/Int/Spring, ALS-L, ENGL`, we only want to store that full combination, rather than storing a new filter combination each time we add a new filter. If we called `updateRecentFilters` during `onChange`, there would be three new recent filter combinations added to the list in the process: `18/19:Fall/Int/Spring`, `18/19:Fall/Int/Spring, ALS-L`, and `18/19:Fall/Int/Spring, ALS-L, ENGL`. Calling it during `onLeave`, we only add the desired complete combination: `18/19:Fall/Int/Spring, ALS-L, ENGL`.